device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/Mqtt.java

```diff
@@ -49,56 +54,57 @@
-    final static String CONTENT_ENCODING = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ce";
-    final static String CREATION_TIME_UTC = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ctime";
-    final static String MQTT_SECURITY_INTERFACE_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ifid";
+    protected final static String CORRELATION_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".cid";
+    protected final static String MESSAGE_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".mid";
+    protected final static String TO = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".to";
+    protected final static String USER_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".uid";
+    protected final static String OUTPUT_NAME = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".on";
+    protected final static String CONNECTION_DEVICE_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".cdid";
+    protected final static String CONNECTION_MODULE_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".cmid";
+    protected final static String CONTENT_TYPE = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ct";
+    protected final static String CONTENT_ENCODING = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ce";
+    protected final static String CREATION_TIME_UTC = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ctime";
+    protected final static String MQTT_SECURITY_INTERFACE_ID = MESSAGE_SYSTEM_PROPERTY_IDENTIFIER_DECODED + ".ifid";
 
     private final static String IOTHUB_ACK = "iothub-ack";
 
     private final static String INPUTS_PATH_STRING = "inputs";
     private final static String MODULES_PATH_STRING = "modules";
 
-    private final IotHubListener listener;
-    private final String connectionId;
+    private IotHubListener listener;
+    private String connectionId;
     private final String deviceId;
 
+    private final Object stateLock; // lock for preventing simultaneous connects and subscribe requests
+    private final Object unacknowledgedSentMessagesLock; // lock for making operations on unacknowledgedSentMessages atomic
+
     /**
      * Constructor to instantiate mqtt broker connection.
-     * @param mqttConnection the connection to use
-     * @param listener the listener to be called back upon connection established/lost and upon a message being delivered
+     * @param mqttAsyncClient the connection to use
      * @param messageListener the listener to be called back upon a message arriving
-     * @param connectionId the id of the connection
      * @param deviceId the Id of the device this connection belongs to
-     * @throws IllegalArgumentException if the provided mqttConnection is null
      */
-    public Mqtt(MqttConnection mqttConnection, IotHubListener listener, MqttMessageListener messageListener, String connectionId, Map<Integer, Message> unacknowledgedSentMessages, String deviceId) throws IllegalArgumentException
+    public Mqtt(
+        MqttAsyncClient mqttAsyncClient,
```

Comment:
Does changing these parameters constitute a breaking change? Class and ctor are public, but I know with Java that isn't a sure way to know what is public interface.

Reply:
Nothing in the MQTT layer is considered part of our public API. Users aren't creating these types

Reply:
We have added and removed params on these classes many times in the past